dist/functions.setup-appliance.sh

```diff
@@ -143,12 +143,27 @@ function get_hostname {
   else
     TIMEOUT=30
     while [ -z "$FQHOSTNAME" -o "$FQHOSTNAME" = "localhost" ];do
-      FQHOSTNAME=`hostname -f 2>/dev/null`
+      # Try to get the FQHN via hostname
+      HN=`hostname -f 2>/dev/null`
+      if [[ $HN =~ ^[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)+$ ]];then
+        FQHOSTNAME=$HN
+      else
+        HN=`hostnamectl --static 2>/dev/null`
+        if [[ $HN =~ ^[A-Za-z0-9-]+(\.[A-Za-z0-9-]+)+$ ]];then
+          FQHOSTNAME=$HN
+        else
```

Comment:
/etc/hostname should be not checked in the loop - only if we run into a timeout.

I`m sorry this was my fault in my suggestion.